taraxa/state/dpos/precompiled/dpos_contract.go

```diff
@@ -472,15 +479,27 @@ func (self *Contract) DistributeRewards(rewardsStats *rewards_stats.RewardsStats
 	}
 }
 
+func (self *Contract) delegate_update_values(ctx vm.CallFrame, validator *Validator, prev_vote_count uint64) {
+	// ctx.Account == contract address. Substract tokens that were sent to the contract as delegation
+	ctx.Account.SubBalance(ctx.Value)
+	validator.TotalStake = bigutil.Add(validator.TotalStake, ctx.Value)
+	self.amount_delegated = bigutil.Add(self.amount_delegated, ctx.Value)
+	new_vote_count := voteCount(validator.TotalStake, self.cfg.EligibilityBalanceThreshold, self.cfg.VoteEligibilityBalanceStep)
+	if prev_vote_count != new_vote_count {
+		self.eligible_vote_count -= prev_vote_count
+		self.eligible_vote_count = add64p(self.eligible_vote_count, new_vote_count)
+	}
+}
+
 // Delegates specified number of tokens to specified validator and creates new delegation object
-// It also increase total stake of specified validator and creas new state if necessary
+// It also increase total stake of specified validator and creates new state if necessary
 func (self *Contract) delegate(ctx vm.CallFrame, block types.BlockNum, args ValidatorAddress) error {
 	validator := self.validators.GetValidator(&args.Validator)
 	if validator == nil {
 		return ErrNonExistentValidator
 	}
 
-	if self.cfg.ValidatorMaximumStake.Cmp(bigutil.Big0) != 0 && self.cfg.ValidatorMaximumStake.Cmp(bigutil.Add(ctx.Value, validator.TotalStake)) == -1 {
+	if self.cfg.ValidatorMaximumStake.Cmp(bigutil.Add(ctx.Value, validator.TotalStake)) == -1 {
```

Comment:
I think `self.cfg.ValidatorMaximumStake.Cmp(bigutil.Big0) != 0` was here as it is optional so 0 means no maxstake limit

Reply:
We have all this checks here https://github.com/Taraxa-project/taraxa-evm/blob/2e558c90275a2e1d94e3ce44e8f19af356b0e4b3/taraxa/state/dpos/precompiled/api.go#L63